packages/app/src/cli/prompts/deploy-release.ts

```diff
@@ -0,0 +1,180 @@
+  release: boolean
+  force: boolean
+}
+
+export interface DeployConfirmationPromptOptions {
+  appTitle?: string
+  extensionsContentPrompt: {
+    extensionsInfoTable?: InfoTableSection
+    deletedInfoTable?: InfoTableSection
+  }
+  configContentPrompt: {
+    configInfoTable: InfoTableSection
+    deletedInfoTable?: InfoTableSection
+  }
+  release: boolean
+}
+
+export async function deployOrReleaseConfirmationPrompt({
+  force,
+  extensionIdentifiersBreakdown,
+  configExtensionIdentifiersBreakdown,
+  appTitle,
+  release,
+}: DeployOrReleaseConfirmationPromptOptions) {
+  if (force) return true
+  const extensionsContentPrompt = await buildExtensionsContentPrompt(extensionIdentifiersBreakdown)
+  const configContentPrompt = await buildConfigContentPrompt(release, configExtensionIdentifiersBreakdown)
+
+  return deployConfirmationPrompt({
+    appTitle,
+    extensionsContentPrompt,
+    configContentPrompt,
+    release,
+  })
+}
+
+async function deployConfirmationPrompt({
+  appTitle,
+  extensionsContentPrompt: {extensionsInfoTable, deletedInfoTable},
+  configContentPrompt: {configInfoTable, deletedInfoTable: configDeletedInfoTable},
+  release,
+}: DeployConfirmationPromptOptions): Promise<boolean> {
+  const timeBeforeConfirmationMs = new Date().valueOf()
+  let confirmationResponse = true
+
+  let question = `Create a new version${appTitle ? ` of ${appTitle}` : ''}?`
+  if (release) {
+    question = `Release a new version${appTitle ? ` of ${appTitle}` : ''}?`
+  }
+
+  let finalDeletedInfoTable = deletedInfoTable ?? {header: '', items: [], bullet: '-'}
+  if (configDeletedInfoTable) {
+    finalDeletedInfoTable = {
+      ...finalDeletedInfoTable,
+      header: configDeletedInfoTable.header,
+      items: [...finalDeletedInfoTable.items, ...configDeletedInfoTable.items],
+    }
+  }
+
+  const infoTable = []
+  if (extensionsInfoTable || finalDeletedInfoTable.header !== '' || configInfoTable.items.length > 0) {
+    infoTable.push(
+      configInfoTable.items.length === 0 ? {...configInfoTable, items: [{subdued: 'No changes'}]} : configInfoTable,
+    )
+  }
+  if (extensionsInfoTable) infoTable.push(extensionsInfoTable)
+  if (finalDeletedInfoTable.header !== '') infoTable.push(finalDeletedInfoTable)
+
+  const isDangerous = appTitle !== undefined && deletedInfoTable
+
+  if (isDangerous) {
+    confirmationResponse = await renderDangerousConfirmationPrompt({
+      message: question,
+      infoTable,
+      confirmation: appTitle,
+    })
+  } else {
+    let confirmationMessage
+
+    if (release) {
+      confirmationMessage = 'Yes, release this new version'
+    } else {
+      confirmationMessage = 'Yes, create this new version'
+    }
+
+    confirmationResponse = await renderConfirmationPrompt({
+      message: question,
+      infoTable,
+      confirmationMessage,
+      cancellationMessage: 'No, cancel',
+    })
+  }
+
+  const timeToConfirmOrCancelMs = new Date().valueOf() - timeBeforeConfirmationMs
```

Comment:
This variable was existing previously so I didn't touch it. Anyway, I would just remove it and create an issue to dig into why it's not used instead of increase the size of this PR